keps/sig-node/20181106-in-place-update-of-pod-resources.md

```diff
@@ -309,7 +334,7 @@ before applying limit increases.
 
 Pod v1 core API:
 * extended model,
-* new subresource,
+* new admission controller,
 * added validation.
 
 Admission Controllers: LimitRanger, ResourceQuota need to support Pod Updates:
```

Comment:
Good point.

From what I can see, the simplest way is to introduce admitPodUpdate method to NodeRestriction plugin that verifies only ResourcesAllocated field is being touched, and that node updating the pod owns the pod. I'll try it out and see if that covers it without leaving any holes.

For authorization, I have modified NodeRules() in plugin/pkg/auth/authorizer/rbac/bootstrappolicy/policy.go to allow nodes to update the pod resource. (They are allowed to create and delete pods at this time). 

The above approach is consistent with how pod creates and deletes by node are handled.

Reply:
Since the proposal is scoped to support only cpu and memory resources, is kubelet only authorized to change those values?  I am assuming that we would want the kubelet to report all resources allocated and enforced (not just cpu and memory), but we would not want to let a user change the pod spec in validation for anything other than cpu and memory?  Is that an accurate understanding?

Reply:
the alternative is that the pod admission plugin sets allocated for all resources other than cpu/memory, but that would make extending this support to other future resource types challenging.

Reply:
Kubelet can continue to report status on all resources. I don't see a need to restrict status changes for the new Resources field. However, for ResourcesAllocated field, it is best to start with allowing Node to change what's actually supported now. As we add support for other resource types, we can just add to the list of supported resource types in the admission plugin.

And yes, for the user, we definitely want to lock it down to just what's supported - cpu and memory